### ISSUE TITLE
Dependency errors do not show `title` or `ui:title`

### ISSUE THREAD

**Issue by chibacchie** [rjsf-team/react-jsonschema-form#4402]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net), [Codesandbox.io](https://codesandbox.io) or preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

validator-ajv8

### Version

5.23.1

### Current Behavior

Dependency errors do not show `title` or `ui:title`.

```javascript
const Ajv2019 = require('ajv/dist/2019');
const { customizeValidator } = require('@rjsf/validator-ajv8');

const schema = {
  type: 'object',
  properties: {
    creditCard: {
      type: 'number',
      title: 'Credit card',
    },
    billingAddress: {
      type: 'string',
      title: 'Billing address',
    },
  },
  dependentRequired: {
    creditCard: ['billingAddress'],
  },
};

const validator = customizeValidator({ AjvClass: Ajv2019 });
const result = validator.validateFormData({ creditCard: 1234567890 }, schema);
console.log(result.errors[0].message);
```

This results in `must have property billingAddress when property creditCard is present`.

### Expected Behavior

`must have property 'Billing address' when property 'Credit card' is present` is expected.

There are two possible causes:

1. `ajv` does not enclose raw property names with quotes (see [here](https://github.com/ajv-validator/ajv/blob/69568d08564303e2c32a2de61feb833b41075f96/lib/vocabularies/applicator/dependencies.ts#L33)) while [transformRJSFValidationErrors()](https://github.com/rjsf-team/react-jsonschema-form/blob/1c443a60922c0d0bf4a652af74399c99bf325907/packages/validator-ajv8/src/processRawValidationErrors.ts#L29) expects property names are enclosed.
2. `transformRJSFValidationErrors()` only handles `missingProperty` (see [this block](https://github.com/rjsf-team/react-jsonschema-form/blob/1c443a60922c0d0bf4a652af74399c99bf325907/packages/validator-ajv8/src/processRawValidationErrors.ts#L40-L55)). `property` and `deps` should be handled as well.

### Steps To Reproduce

Please see above.

### Environment

```markdown
- OS: Ubuntu 24.04
- Node: 18.20.2
- npm: 10.5.0
```

**Comment by heath-freenome** [comment]
@chibacchie does your fix in https://github.com/rjsf-team/react-jsonschema-form/pull/4398 also fix this issue?

**Comment by chibacchie** [comment]
@heath-freenome I did not fix this issue in #4398. To fix this issue, we may:

1. Ask `ajv` to enclose raw property names with quotes
2. Improve `transformRJSFValidationErrors()` to handle `property` and `deps`

**Comment by heath-freenome** [comment]
@chibacchie The probability of AJV making that change is lower than fixing our code. Is that something that you are willing to attempt?

**Comment by chibacchie** [comment]
@heath-freenome Indeed. I made a PR which is only applicable if we use an ajv-18n localizer (See #4417). For the case without localizer, I plan to submit a PR for ajv as I don't have any idea to fix this issue without chaning ajv.

**Comment by TheOneTheOnlyJJ** [comment]
I agree, we should do our best to request consistency from ajv.

**Comment by heath-freenome** [comment]
@chibacchie did you ever file that PR with AJV?
